cpp/src/arrow/array/array_list_test.cc

```diff
@@ -508,6 +534,8 @@ TYPED_TEST(TestListArray, ValidateOffsets) { this->TestValidateOffsets(); }
 
 TYPED_TEST(TestListArray, CornerCases) { this->TestCornerCases(); }
 
+TYPED_TEST(TestListArray, TestOverflowCheck) { this->TestOverflowCheck(); }
```

Comment:
```suggestion
TYPED_TEST(TestListArray, LARGE_MEMORY_TEST(TestOverflowCheck)) { this->TestOverflowCheck(); }
```

Reply:
It doesn't seem to work with TYPED_TEST macro, so I expanded it manually.